egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,299 @@
+                wav_id = prefix + str(i) + "_" + wav_id
+            command = "{0} {1} wav-reverberate".format(wav_id, wav_pipe)
+
+            # pick the room
+            room = rooms.next()
+            command_opts = ""
+            noises_added = []
+            snrs_added = []
+            start_times_added = []
+            if random.random() < speech_rvb_probability:
+                # pick the RIR to reverberate the speech
+                speech_rir = room['rir_list'][random.randint(0,len(room['rir_list'])-1)]
+                command_opts += "--impulse-response={0} ".format(speech_rir.rir_file_location)
+                # add the corresponding isotropic noise if there is any
+                if len(speech_rir.iso_noise_list) > 0:
+                    isotropic_noise = speech_rir.iso_noise_list[random.randint(0,len(speech_rir.iso_noise_list)-1)]
+                    noises_added.append("{0}".format(isotropic_noise.noise_file_location))
+                    snrs_added.append("{0}".format(snrs.next()))
+                    start_times_added.append(round(random.random() * wav_dur, 2))
+
+            if noises is not None and random.random() < noise_adding_probability:
+                for k in range(random.randint(1, max_noises_added)):
+                    # pick the RIR to reverberate the point-source noise
+                    noise = noises.next()
+                    noise_rir = room['rir_list'][random.randint(0,len(room['rir_list'])-1)]
+                    start_times_added.append(round(random.random() * wav_dur, 2))
+                    noises_added.append("\"wav-reverberate --duration={2} --impulse-response={1} {0} - |\" ".format(noise.noise_file_location, noise_rir.rir_file_location, round(random.random()*(wav_dur-start_times_added[-1]), 2)))
+                    snrs_added.append("{0}".format(snrs.next()))
+
+            if len(noises_added) > 1:
+                command_opts += "--additive-signals='{0}' ".format(','.join(noises_added))
+            if len(snrs_added) > 1:
+                command_opts += "--snrs='{0}' ".format(','.join(snrs_added))
+            if len(start_times_added) > 1:
+                command_opts += "--start-times='{0}' ".format(','.join(snrs_added))
+            
+            if command_opts == "":
+                command = "{0} {1}\n".format(wav_id, wav_pipe) 
+            else:
+                command = "{0} {1} wav-reverberate {2} - - |\n".format(wav_id, wav_pipe, command_opts)
+
+            command_list.append(command)
+
+    file_handle = open(output_dir + "/wav.scp", 'w')
+    file_handle.write("".join(command_list))
+    file_handle.close()
+
+
+# This function replicate the entries in files like text
+def ReplicateFileType1(input_file, output_file, num_replica, prefix):
+    list = map(lambda x: x.strip(), open(input_file))
+    f = open(output_file, "w")
+    for i in range(num_replica):
+        for line in list:
+            split1 = line.split()
+            if prefix is not None:
```

Comment:
I think it is better do add prefix at all times. You will never by reverberating test sets and even if you do, you should add the prefix to appropriate columns in both reco2file_and_channel and stm. GLM should not change because it has nothing to do with the file ids.